mcs/tools/linker/Makefile

```diff
@@ -2,21 +2,40 @@ thisdir = tools/linker
 SUBDIRS =
 include ../../build/rules.make
 
-RESOURCES = \
-	Descriptors/mscorlib.xml	\
-	Descriptors/System.xml		\
-	Descriptors/System.Core.xml		\
-	Descriptors/System.Drawing.xml	\
-	Descriptors/System.Web.xml		\
-	Descriptors/Mono.Posix.xml		\
-
 PROGRAM = monolinker.exe
 
-$(PROGRAM): $(RESOURCES)
-
 LIB_REFS = System System.Core System.Xml Mono.Cecil
-LOCAL_MCS_FLAGS = $(RESOURCES:%=-resource:%)
 
-EXTRA_DISTFILES = $(RESOURCES)
+TEST_CASES = \
+	mscorlib/test-array.cs
+
+ifndef AOT_FRIENDLY_PROFILE
+TEST_CASES += \
+	mscorlib/test-reflection.cs
+endif
+
+TESTS_COMPILER = $(MCS) -nologo -noconfig -debug:portable -r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll
+
+check: compile-tests run-tests
+
+compile-tests: $(TEST_CASES)
+
+mscorlib/test-%.cs:
```

Comment:
But does it do proper dep tracking? IE, things won't be compiled when no change is made.